pkg/webhooks/generate/generate.go

```diff
@@ -35,15 +38,18 @@ type Generator struct {
 	client *kyvernoclient.Clientset
 	stopCh <-chan struct{}
 	log    logr.Logger
+	// grLister can list/get generate request from the shared informer's store
+	grLister kyvernolister.GenerateRequestNamespaceLister
 }
 
 // NewGenerator returns a new instance of Generate-Request resource generator
-func NewGenerator(client *kyvernoclient.Clientset, stopCh <-chan struct{}, log logr.Logger) *Generator {
+func NewGenerator(client *kyvernoclient.Clientset, grInformer kyvernoinformer.GenerateRequestInformer, stopCh <-chan struct{}, log logr.Logger) *Generator {
 	gen := &Generator{
-		ch:     make(chan GeneratorChannel, 1000),
-		client: client,
-		stopCh: stopCh,
-		log:    log,
+		ch:       make(chan GeneratorChannel, 1000),
+		client:   client,
+		stopCh:   stopCh,
+		log:      log,
+		grLister: grInformer.Lister().GenerateRequests(config.KyvernoNamespace),
```

Comment:
In order to use the lister, you have to wait till the cache is synced on start. For example:
https://github.com/kyverno/kyverno/blob/55d7c34fdf18d9c3f92c03f54179bd197baa8f6a/pkg/generate/generate_controller.go#L111-L115
https://github.com/kyverno/kyverno/blob/55d7c34fdf18d9c3f92c03f54179bd197baa8f6a/pkg/generate/generate_controller.go#L261